### DISCUSSION TITLE
Environments should also contain non-secrets

### DISCUSSION THREAD

**Discussion by newhoggy** [actions/runner#1027] [Ideas]
Currently environments can be used to pass variables to jobs, but all variables must be secret.  This makes the user experience very poor for settings I want to pass to jobs that are no secrets.  For example, download URLs or number of threads. 

Sometimes the value is so short, making it a secret hurts usability.  For example, if I wanted to pass the value `1` into my job, then everything that is `1` in the build output gets masked out, which is quite bad.

**Comment by saeidasade321** [comment]
- [](url)
[fpm-1.12.0.zip](https://github.com/actions/runner/files/6477482/fpm-1.12.0.zip)
